### PR TITLE
chore: add notification interface and move email auth to config

### PR DESCRIPTION
**Pull request by markkovari**
Sorry for the PR only, but I have pretty much no time for real-time talk this week. The code looks good, but we should - in my opinion- centralize the configs and make the email an instance of a notificationService.

### CODE REVIEW

File: sk0-back/src/notifications/mailer.service.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+{
+  private transporter: Transporter;
+  constructor(private readonly configService: ConfigService) {
+    const auth = {
+      user: this.configService.get<string>("EMAIL_USER"),
+      pass: this.configService.get<string>("EMAIL_PASSWORD")
+    };
+    if (!auth.user || !auth.pass) {
+      throw new Error("Email credentials not set");
+    }
```
Comment (markkovari) [inline sk0-back/src/notifications/mailer.service.ts]: Not sure if we can/should push this to the config level or not? 🤔

Comment (markkovari) [reply sk0-back/src/notifications/mailer.service.ts]: ooooh, there is a getOrThrow for fields too, it might be shorter :P

**Comment by rolcho** [comment]
Thanks I will walk through to see what have been changed. You wrote that there should be a common notificaton service. Does it means that we should (it could be worth to) implement a Delegate pattern?

**Comment by markkovari** [comment]
Yes we can do that, not sure how dynamic this will be, but makes sense.We might even wanted to make a feature flag for now? Not 100%

File: sk0-back/src/notifications/mailer.service.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+@Injectable()
+export class MailerService
+  implements NotificationService<MailerSuccessResponse>
+{
+  private transporter: Transporter;
+  constructor(private readonly configService: ConfigService) {
+    const auth = {
+      user: this.configService.get<string>("EMAIL_USER"),
+      pass: this.configService.get<string>("EMAIL_PASSWORD")
```
Comment (rolcho) [inline sk0-back/src/notifications/mailer.service.ts]:
```suggestion
      user: this.configService.getOrThrow<string>("EMAIL_USER"),
      pass: this.configService.getOrThrow<string>("EMAIL_PASSWORD")
```

File: sk0-back/src/notifications/mailer.service.ts (hunk 1 above)
Comment (rolcho) [inline sk0-back/src/notifications/mailer.service.ts]: You are right! and we can remove these lines as well
```suggestion
```

**Review by rolcho — COMMENTED** [review]
:sunglasses:

**Review by rolcho — APPROVED** [review]

**Pull request merged**